src/main/java/com/atlassian/db/replica/internal/RouteDecisionBuilder.java

```diff
@@ -0,0 +1,56 @@
+package com.atlassian.db.replica.internal;
+
+import com.atlassian.db.replica.api.context.Reason;
+import com.atlassian.db.replica.api.context.RouteDecision;
+
+import java.util.Objects;
+
+public final class RouteDecisionBuilder {
+    private Reason reason;
+    private String sql;
+
+    public RouteDecisionBuilder(Reason reason) {
+        this.reason = reason;
+    }
+
+    public RouteDecisionBuilder reason(final Reason reason) {
+        this.reason = reason;
+        return this;
+    }
+
+    public RouteDecisionBuilder sql(final String sql) {
+        this.sql = sql;
+        return this;
+    }
+
+    public String getSql() {
+        return sql;
+    }
+
+    public RouteDecision build() {
```

Comment:
This builder doesn't provide defaults. It's used to override the `reason` or optionally attach an `sql`.
We could reduce boilerplate by removing the builder and adding `RouteDecision withReason(Reason reason)` to `RouteDecision` itself. Same for `withSql(String sql)`.

Unless we need to distinguish "partial" decision from a full decision explicitly.